src/attr/__init__.pyi

```diff
@@ -470,7 +470,15 @@ def astuple(
     tuple_factory: Type[Sequence[Any]] = ...,
     retain_collection_types: bool = ...,
 ) -> Tuple[Any, ...]: ...
-def has(cls: type) -> bool: ...
+
+if sys.version_info >= (3, 10):
```

Comment:
Would it be feasible to add a conditional import from typing-extensions?

Reply:
You can only use one of a small number of constants like `TYPE_CHECKING` and `sys.version_info` to define types conditionally, try/except won't work, so I don't think so.

Reply:
In my very limited testing this seems to work, both in Mypy and Pyright:

<img width="422" alt="Screenshot 2022-08-15 at 13 37 05" src="https://user-images.githubusercontent.com/1909233/184628381-d79a5c89-f1f2-4f2f-8a39-d683ac593ab9.png">

Reply:
Right, but typing_extensions is not a dependency of attrs, so that'll error if typing_extensions is not installed.

Reply:
Okay, it seems typeshed ships typing_extensions is part of the stdlib, so I guess we can actually do that.

Reply:
Done in https://github.com/python-attrs/attrs/pull/997/commits/fd2c819ab04bbfe4785ee6abbc9c42a8f566dc22.